openghg/util/_platform.py

```diff
@@ -1,5 +1,105 @@
+from typing import overload
+import logging
 from openghg.util._site import get_site_info
-from openghg.types import pathType
+from openghg.util._metadata_util import not_set_metadata_values
+from openghg.types import pathType, MetadataFormatError
+
+logger = logging.getLogger("openghg.util")
+logger.setLevel(logging.DEBUG)  # Have to set level for logger as well as handler
+
+
+def define_platform(data_type: str | list | None = None) -> list:
+    """
+    Define platform values which can be used. These describe the type of data and
+    platform used to measure the data.
+
+    Args:
+        data_type: Specify the data_type to select the associated platforms.
+            For "surface" this is currently:
+                - "surface-insitu"
+                - "surface-flask"
+            For "column" this is currently:
+                - "satellite"
+                - "column-insitu"
+                - "column"
+            For "mobile" this is currently:
+                - "aircraft-flask"
+                - "balloon-flask"
+                - "mobile-flask"
+            If no data_type is specified, this will return all platform values.
+    Returns:
+        list: Platform values
+    """
+
+    platform_values = []
+
+    platform_values_all = {
+        "surface": [
+            "surface-insitu",
+            "surface-flask",
+        ],
+        "column": [
+            "satellite",
+            "column-insitu",
+            "column",
+        ],
+        "mobile": [
+            "aircraft-flask",
+            "balloon-flask",
+            "mobile-flask",
+        ],
+    }
+
+    obs_types = ["surface", "column", "mobile"]
+
+    if data_type is None:
+        data_types = obs_types
+    elif isinstance(data_type, str):
+        data_types = [data_type]
+    else:
+        data_types = data_types
+
+    for data_type in data_types:
+        platform_values.extend(platform_values_all[data_type])
+
+    return platform_values
+
+
+@overload
+def format_platform(platform: str, data_type: str | None = None) -> str: ...
+
+
+@overload
+def format_platform(platform: None, data_type: str | None = None) -> None: ...
+
+
+def format_platform(platform: str | None, data_type: str | None = None) -> str | None:
+    """
+    Check platform is a suitable value based on the values within define_platform.
```

Comment:
```suggestion
    Check platform is a suitable value based on the values within define_platform.
    This could change the case of the platform value to match to the define_platform values if the original case does not match.
```